src/transaction/log_applier_sql_log.c

```diff
@@ -612,44 +625,124 @@ sl_open_next_file (FILE * old_fp)
   return new_fp;
 }
 
+/*
+ * sl_remove_oldest_file() - remove oldest sql log file
+ *   return: error code
+ *
+ * Note:
+ *   This function is related to the ha_sql_log_max_count system parameter.
+ *   This system parameter can be set from 2 to 5 and only that number of sql log files are kept.
+ */
+static int
+sl_remove_oldest_file (void)
+{
+  int oldest_file_id;
+  char oldest_file_path[PATH_MAX];
+
+  oldest_file_id = sl_Info.curr_file_id - sql_log_max_cnt;
+
+  snprintf (oldest_file_path, PATH_MAX - 1, "%s.%d", sql_log_base_path, oldest_file_id);
+
+  unlink (oldest_file_path);
+
+  return NO_ERROR;
+}
+
+/*
+ * sl_create_sql_log_dir() - verify and create the SQL log path
+ *   return: NO_ERROR or ER_FAILED
+ *     repl_log_path(in): log volume path for apply (default path)
+ *     path_buf(out): SQL log path buffer
+ *     path_buf_size(in): buffer size
+ *
+ * Note:
+ *   This function is related to the ha_sql_log_path system parameter. The SQL log path can be changed by setting this.
+ */
 static int
-create_dir (const char *new_dir)
+sl_create_sql_log_dir (const char *repl_log_path, char *path_buf, int path_buf_size)
 {
-  char *p, path[PATH_MAX];
+  int path_len;
+  const char *log_path = NULL;
+  char *p = NULL;
+  char er_msg[PATH_MAX];
+
+  assert (repl_log_path != NULL && path_buf != NULL && path_buf_size >= PATH_MAX);
 
-  if (new_dir == NULL)
+  log_path = prm_get_string_value (PRM_ID_HA_SQL_LOG_PATH);
+  if (log_path != NULL && *log_path != '\0')
     {
-      return ER_FAILED;
+      if (!IS_ABS_PATH (log_path))
+	{
+	  snprintf (path_buf, path_buf_size, "%s%s%s", repl_log_path, FILEIO_PATH_SEPARATOR (repl_log_path), log_path);
+
+	  log_path = path_buf;
+	}
+    }
+  else
+    {
+      log_path = repl_log_path;
+    }
+
+  if (log_path == path_buf)
+    {
+      path_len = strlen (log_path);
+
+      snprintf (path_buf + path_len, path_buf_size - path_len, "%s%s", FILEIO_PATH_SEPARATOR (log_path), "sql_log");
+    }
+  else
+    {
+      snprintf (path_buf, path_buf_size, "%s%s%s", log_path, FILEIO_PATH_SEPARATOR (log_path), "sql_log");
     }
 
-  strcpy (path, new_dir);
+  if (strlen (path_buf) >= path_buf_size - 1)
```

Comment:
This won't happen since `snprintf` cut the argument's string and attach the NULL at the end of string if the size of format exceeds the given size.

Reply:
I guess this error handling should be placed before snprintf (path_buf, path_buf_size, "%s%s%s", log_path, FILEIO_PATH_SEPARATOR (log_path), "sql_log") with the calculated size;

Reply:
I think that '>' makes you confusing. I wanted to verify the case where the path string created in path_buf exceeded PATH_MAX and was truncated.